src/Services/SodiumService.php

```
@@ -139,7 +140,7 @@ public function decryptValueByNonce(string $value, string $nonce): ?string
     {
         try {
             return $this->decrypt($nonce ? implode('.', [$nonce, $value]) : $value);
-        } catch (DecryptException | KeyNotFoundException | MalformationException $e) {
+        } catch (DecryptException | KeyNotFoundException | MalformationException | SodiumException $e) {
```

I think swallowing all these exceptions is not appropriate here. We should let these to thrown and the user of this package should deal with the Exceptions. For example, making this change will break STDCheck logging unencrypted value on Model . `app/Model/EncryptedModel.php:276`

And of the projects that are using this Sodium package should approach this differently. For example, if the data coming from a Request then we can validate them to filter the malformed values. And for the Model side, we should always keep logging the malformed information to be alert to prevent patients data to be lost.

Creating all the validations on STDCheck and other projects is out of the scope of this issue. But issue can be created after some investigation

---

Totally agreed with you.